sql-plugin/src/main/java/com/nvidia/spark/rapids/InternalRowToColumnarBatchIterator.java

```diff
@@ -186,25 +186,25 @@ public ColumnarBatch next() {
         }
       } catch (BufferOverflowException ex) {
         // Handle corner case when the dataLength is too small to copy a single row.
-        // Increase dataBuffer size by 25%
+        // We retry after doubling the bufferSize.
         // dataLength can be considered a rough estimate of a single row.
-        long newRowSizeEstimate =
-            Math.min((dataLength * 125) / 100, JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH);
-        if (newRowSizeEstimate <= dataLength) { // We already reached the limit.
+        long newRowSizeEst = dataLength << 1 ;
+        newRowSizeEst = Math.min(newRowSizeEst, JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH);
```

Comment:
```suggestion
        long newRowSizeEst = Math.min(dataLength << 1, JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH);
```
It looks odd to me to write to newRowSizeEst and then immediately assign to it again. This got me looking more at this code and now I have questions too.

So we have a buffer of size dataLength and it is too small, so we double the buffer size in newRowSizeEst and then set it to the min of that doubled buffer size and a max. Then we compare the new size we want to make the buffer to the old buffer size and if it is smaller or the same, we bail. The only time this could happen is if dataLength is currently `JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH`, right? If that is the case, why won't we just check that? This way seems a little convoluted.

```
        if (dataLength == JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH) { // this won't work...
        // double buffer size and try again
        long newRowSizeEst = Math.min(dataLength << 1, JCudfUtil.JCUDF_MAX_DATA_BUFFER_LENGTH);
```

Reply:
Ok sure. Easy fix.